src/fs/filesystem.ts

```diff
@@ -273,17 +286,34 @@ export class FileSystem implements UnixTree {
   }
 
 
+  // PUBLICIZE
+  // ---------
+
   /**
-   * Ensures the latest version of the file system is added to IPFS and returns the root CID.
+   * Ensures the latest version of the file system is added to IPFS,
+   * updates your data root, and returns the root CID.
    */
-  async publicize(): Promise<CID> {
+  async publicise(): Promise<CID> {
     const cid = await this.root.put()
+    const proofs = [...this.proofs]
 
-    this.syncHooks.forEach(hook => hook(cid))
+    this.proofs = []
+
+    proofs.forEach(proof => {
```

Comment:
Wanna make sure I'm following the logic correctly

We're storing 1 proof for each change to the file system between publishing & then sending an "udpate data root" req for each proof but with the same CID, right?

In order to process a request, wouldn't the server need all of the proofs for a given change? Together, the proofs are sufficient to change the root CID, but alone I think that each would be insufficient.

Also do we want to perform some sort of dedup here? For instance if a user has `"*"` permission & makes 20 updates before `publicise`ing, we don't need to send 20 requests

Reply:
> We're storing 1 proof for each change to the file system between publishing & then sending an "udpate data root" req for each proof but with the same CID, right?
Yes

> In order to process a request, wouldn't the server need all of the proofs for a given change?
I think a single change only needs one proof, no?
That's what I gathered from what Brooke told me.
But I could be totally wrong on this.
I don't entirely understand how the server would validate a filesystem/dnslink UCAN 🤔 

> Also do we want to perform some sort of dedup here?
Yeah good point, will add this 👍

Reply:
> I think a single change only needs one proof, no?

It depends....
Imagine that I have permission for `private/Music` & `private/Photos`, but nothing else. Locally I add something to each folder & then try to sync the CID. 

I think the code right now is submitting two requests:
- req 1 contains the updated CID & a UCAN that proves access to modify `private/Music`
- req 2 contains the updated CID & a UCAN that proves access to modify `private/Photos`

But for the server, this is a single transaction & it needs proof that you can modify both. So we oughta be sending:
- a single req with the updated CID & a UCAN that proves access to modify both `private/Music` & `private/Photos`

Does that make sense/match your understanding?

Reply:
> But for the server, this is a single transaction & it needs proof that you can modify both.

I'm not sure I understand why this would be transformed into a single transaction 🤔 
Why not separate? cc @expede

Reply:
Oh wait I think I understand. It's a single transaction because we do `cid = await this.root.put()` in the sdk and then update the data root with that cid?

Reply:
Ok so, can we do this with separate UCANs @expede ? Or do we need a UCAN with multiple scopes (ie. UCAN v1.1)?

Reply:
> It's a single transaction because we do cid = await this.root.put() in the sdk and then update the data root with that cid?

Yup exactly. that `cid` is representative of a Filesystem that contains _both_ updates. So in order to update the user's data root to the new `cid`, you need to prove that you have the permission to do both updates.

> Ok so, can we do this with separate UCANs @expede ? Or do we need a UCAN with multiple scopes (ie. UCAN v1.1)?

Both would work I believe, but my understanding is that UCAN's with multiple scopes are targeted at solving just this issue.  Actually I'm gonna go back & read some of the talk forum posts, I remember a discussion on the tradeoffs here. Reviewing this PR has been really for me in getting UCAN stuff more solid in my brain :+1:

Reply:
Here’s the spec: https://app.gitbook.com/@runfission/s/whitepaper/access-control/ucan/jwt-authentication

Reply:
> Ok so, can we do this with separate UCANs @expede ?

In theory yes, but we won’t be supporting it, since it’s a subset of what’s possible with UCAN 0.4

Reply:
I'll address the UCAN v0.4 changes in another PR.
The foundation for those changes is now there.

The auth lobby will return a UCAN with the `*` rsc for now, so it works with this SDK version and the current API.